rlclientlib/live_model_impl.cc

```diff
@@ -604,6 +627,45 @@ namespace reinforcement_learning {
     return error_code::success;
   }
 
+  int reset_action_order_multi_slot(multi_slot_response& response, const std::vector<int>& baseline_actions)
+  {
+    int index = 0;
+    for (auto &slot : response)
+    {
+      if (baseline_actions.size() != 0 && baseline_actions.size() >= index)
+      {
+        slot.set_action_id(baseline_actions[index]);
+      }
+      else
+      {
+        //implicit baseline is the action corresponding to the slot index
+        slot.set_action_id(index);
+      }
+      slot.set_probability(1);
```

Comment:
why is the probability set to 1 here?

Reply:
When we are in apprentice/logging only mode, the probability of returning the baseline action (either implicit or explicit) is 1.

Reply:
But do we want to have the same change for cb then?

Reply:
I explained in the description why we decided against it for the cb case. 

Some clients use the probabilities to decide on actions to populate in a "carousel". This is where not having a probability associated with the recommended action may be problematic. In apprentice mode we promise that the user can move to online with a "flip of a switch" so we cannot have different functionality for apprentice and online mode.

So basically in the cb case, some customers in apprentice mode put baseline action in the first "slot" of the carousel and then populate the rest of the carousel based on the probabilities in descending order. If we change the probabilities so that baseline action has a probability of 1 and the rest 0, this will break our promise that they can move to online mode with a "flip of a switch".

In the multi_slot_response case the customers do not have the PDF anymore (they only have recommended action and its associated probability) so this case scenario does not apply.